build.rs

```diff
@@ -1,73 +1,34 @@
-use std::io::Result;
+use prost_types::{FileDescriptorProto, FileDescriptorSet};
+use prost::Message;
+
+fn main() -> std::io::Result<()> {
+    let protos = include_str!(concat!(env!("CARGO_MANIFEST_DIR"), "/protos.list"))
+        .lines()
+        .collect::<Vec<&str>>();
+
+    prost_build::Config::new()
+        // should probably switch to this
+        //.btree_map(&["."])
+        .out_dir("./out")
```

Comment:
It's fine for now, but we shouldn't output outside `OUT_DIR` set by `cargo` in build scripts (`out/` was copied from there for reference).

> Build scripts may save any output files in the directory specified in the `OUT_DIR` environment variable. Scripts should not modify any files outside of that directory.
> https://doc.rust-lang.org/cargo/reference/build-scripts.html#outputs-of-the-build-script

Reply:
How should we copy this effectively if we don't hardcode the output path? The [`cargo build --out-dir` flag is unstable](https://github.com/rust-lang/cargo/issues/6790) (and by the looks of things [is unlikely to get stabilised soon](https://github.com/rust-lang/cargo/issues/6100#issuecomment-821022329)), so don't see how we can effectively piggy-back on top of `OUT_DIR`.

With `OUT_DIR`, the output happens to be within `target/debug/build/k8s-pb-790fe4bcffd42aa5/out/api.core.v1.rs` in my case, but it's a bit awkward to grab from. Should we write the "final output" at the end of the `build.rs` script instead?

Reply:
I think we should have `k8s-pb-codegen` binary crate, like `k8s-openapi-codegen`. This is why I thought `codegenrs` (#3) might be useful. It seems to make it easy to set this up. I haven't looked into it, though.

Reply:
Ok, yeah, we definitely need another place to keep most of the logic at the very least. Tried a little bit to build on top of build.rs and the debug flow there is hard. I'll try to look more at it tomorrow :zzz: